src/test/regress/sql/dml_recursive.sql

```diff
@@ -1,3 +1,9 @@
+-- Two alternative test outputs:
+-- dml_recursive_0.out for PG16 and before
+-- dml_recursive.out for PG17
+-- related commit
+-- PostgreSQL 17 includes an enhancement that allows the optimizer to transform correlated IN subqueries into more efficient join operations.
```

Comment:
Let's hold off a bit more before merging this, until we are sure that nothing needs to change in the Citus codebase.